### PR TITLE
[SYCL][Doc] Update sub-group docs

### PR DESCRIPTION
**Pull request by Pennycook**
Reverts some changes made when switching from SubGroupNDRange to SubGroup extension:
- `max_sub_group_size` must take a work-group size in order to support OpenCL
- The sub-group `barrier` member function was removed by mistake
- The sub-group `shuffle` functions should be supported in addition to the higher-level `permute` and `shift_*` functions from the SubGroupAlgorithms extension

---
Thanks to @AlexeySachkov and @jasonsewall-intel for noticing some of these issues!

### CODE REVIEW

**Comment by Pennycook** [comment]
I hadn't realized that SYCL_device_specific_kernel_queries had been merged in https://github.com/intel/llvm/pull/1540.  I've pushed a new commit to align the sub-group queries with the syntax there -- @felipe-intel, could you please take a look?

**Comment by felipe-intel** [comment]
> 
> 
> I hadn't realized that SYCL_device_specific_kernel_queries had been merged in #1540. I've pushed a new commit to align the sub-group queries with the syntax there -- @felipe-intel, could you please take a look?

LGTM!

**Comment by garimagu** [comment]
@AlexeySachkov  Ping.

File: sycl/doc/extensions/SubGroup/SYCL_INTEL_sub_group.asciidoc [hunk 1]
```diff
@@ -111,33 +111,37 @@ The device descriptors below are added to the +info::device+ enumeration class:
...
-An additional query for sub-group information is added to the +kernel+ class:
+An additional query is added to the +kernel+ class, enabling an input value to be passed to `get_info`:
 
 |===
 |Member Functions|Description
 
-|+template <info::kernel_sub_group param>typename info::param_traits<info::kernel_sub_group, param>::return_type get_sub_group_info(const device &dev) const+
-|Query information from the sub-group from a kernel using the +info::kernel_sub_group+ descriptor for a specific device.
+|+template <info::kernel_device_specific param>typename info::param_traits<info::kernel_device_specific, param>::return_type get_info(const device &dev, typename info::param_traits<info::kernel_device_specific, param>::input_type value) const+
+|Query information from a kernel using the +info::kernel_device_specific+ descriptor for a specific device and input parameter.  The expected value of the input parameter depends on the information being queried.
```
Comment (garimagu) [inline sycl/doc/extensions/SubGroup/SYCL_INTEL_sub_group.asciidoc:120]: Is it required to be mentioned that the second argument is optional when not needed?
The previous version had both versions of get_sub_group_info APIs mentioned.
https://github.com/intel/llvm/blob/fba2e0602550a86c74149d9875b788ad1117f8d3/sycl/doc/extensions/SubGroupNDRange/SubGroupNDRange.md

Comment (Pennycook) [reply sycl/doc/extensions/SubGroup/SYCL_INTEL_sub_group.asciidoc:120]: I've added a reference to SYCL_device_specific_kernel_queries and highlighted that which version should be used depends on whether there is an input type specified for the query or not.  Thanks!

**Comment by garimagu** [comment]
Hi John,
I'm still a little confused about the behavior of this API. 
The API will return max num of sub-groups for a Level0 device and the min(range, max num of sub_groups) for an OpenCL device. Is that correct?

**Comment by jbrodman** [comment]
> Hi John,
> I'm still a little confused about the behavior of this API.
> The API will return max num of sub-groups for a Level0 device and the min(range, max num of sub_groups) for an OpenCL device. Is that correct?

@Pennycook

**Comment by garimagu** [comment]
> > Hi John,
> > I'm still a little confused about the behavior of this API.
> > The API will return max num of sub-groups for a Level0 device and the min(range, max num of sub_groups) for an OpenCL device. Is that correct?
> 
> @Pennycook

I had a call with John. He confirmed that my understanding was correct.
-Garima

**Comment by garimagu** [comment]
@Pennycook Is this PR not merged for some reason. The PR corresponding to this change in SYCL : https://github.com/intel/llvm/pull/1600 is ready to be merged.

**Comment by Pennycook** [comment]
> @Pennycook Is this PR not merged for some reason. The PR corresponding to this change in SYCL : #1600 is ready to be merged.

It cannot be merged without a review from one of the code owners for the extensions directory.  @jbrodman, can you approve if there are no other concerns?  @garimagu and I agree that what's specified here is correct, and we'll open a new issue if future updates are needed.

**Comment by bader** [comment]
@jbrodman, @mkinsner, ping.

**Pull request merged**